monix-execution/jvm/src/main/scala/monix/execution/schedulers/SchedulerCompanionImpl.scala

```diff
@@ -473,7 +480,7 @@ private[execution] class SchedulerCompanionImpl extends SchedulerCompanion {
     implicit lazy val global: Scheduler =
       AsyncScheduler(
         DefaultScheduledExecutor,
-        ExecutionContext.Implicits.global,
+        workStealingThreadPool,
```

Comment:
Don't change the default for now, because `Scheduler.global` is meant for `Future` too, we need to think about the implications.

This is also why I opened #1438 — if we changed the default to a limited fixed thread-pool, maybe we should warn people that still execute `blocking` tasks on top.

Reply:
Oh, I only changed this to try running everything in the CI.

Reply:
I think you need to explicitly allow this CI run since I'm a first time contributor to the project. New GitHub policy.

Reply:
Done — the tests are now running.

Reply:
Do you have some tests for it BTW? Would be nice.

Reply:
I've been meaning to write jcstress tests for the queues. I'm not sure how to test the pool itself. I'm open to suggestions.

Reply:
IDK, some tests involving concurrency, happy path would be better than nothing. You might trigger the execution of some tasks via `sequence` and `parSequence` and see if the result is correct.

Reply:
Note that usage of `Future` in those tests is fine.